Structure_Engine/Query/TorsionalConstant.cs

```diff
@@ -355,57 +418,152 @@ public static double ITorsionalConstant(this IProfile profile)
         /**** Private Methods - helper methods          ****/
         /***************************************************/
 
-        [Description("Diameter of an circles inscribed in a T-junction connection where tf is assumed to be the thickness of the top of the T. Taken from 'P385 Design of steel beams in torsion', Appendix B.")]
+        [Description("Diameter of an circles inscribed in a T-junction connection such as in a parallel flange I-Section. Taken from 'P385 Design of steel beams in torsion', Appendix B.")]
         [Input("tw", "Web thickness, assumed to be the stem of the T.", typeof(Length))]
         [Input("tf", "Flange thickness, assumed to be the top of the T.", typeof(Length))]
         [Input("r", "Root radius, assumed to be the same on both sides of the T.", typeof(Length))]
-        private static double InscribedDiameterTJunction(double tw, double tf, double r)
+        public static double InscribedDiameterTJunction(double tw, double tf, double r)
         {
+            //Equation 23
             return (Math.Pow(tf + r, 2) + (r + 0.25 * tw) * tw) / (2 * r + tf);
         }
 
         /***************************************************/
 
-        [Description("Diameter of an circles inscribed in a L-junction connection. Taken from 'P385 Design of steel beams in torsion', Appendix B.")]
+        [Description("Diameter of an circles inscribed in an L-junction connection such as in an L-Section. Taken from 'P385 Design of steel beams in torsion', Appendix B.")]
         [Input("tw", "Web thickness.", typeof(Length))]
         [Input("tf", "Flange thickness.", typeof(Length))]
         [Input("r", "Root radius.", typeof(Length))]
-        private static double InscribedDiameterLJunction(double tw, double tf, double r)
+        public static double InscribedDiameterLJunction(double tw, double tf, double r)
         {
+            //Equation 25
             return 2 * ((3 * r + tw + tf) - Math.Sqrt(2 * (2 * r + tw) * (2 * r + tf)));
         }
 
         /***************************************************/
 
-        [Description("Emperical formula used to correct the torsional constant with enhancement from a T-junction. Taken from 'P385 Design of steel beams in torsion', Appendix B.")]
+        [Description("Diameter of a circle inscribed in a tapered flange T-junction connection such as a taper flange I-section. Taken from Johnston & El Darwish, 1964, Figure 6 Case 2")]
+        [Input("tw", "Web thickness, assumed to be the stem of the T.", typeof(Length))]
+        [Input("t3", "Flange thickness at theoretical intersection of flange and web centerline.", typeof(Length))]
+        [Input("r", "Root radius, assumed to be the same on both sides of the T.", typeof(Length))]
+        [Input("s", "Flange taper slope.", typeof(Ratio))]
+        public static double InscribedDiameterTaperTJunction(double tw, double t3, double r, double s)
+        {
+            if (s < 0.001) return InscribedDiameterTJunction(tw, t3, r);
```

Comment:
Could make this tolerance significantly smaller, even if that will not matter that much ofc. Maybe use `Tolerance.Angle` instead (I know it is not an angle, but is the closest). Good to avoid using hardcoded tolerances like this unless we have to, Good to also put a short comment why it is done if we come back to this later as well.

Reply:
```suggestion
            if (s < Tolerance.Angle) return InscribedDiameterTJunction(tw, t3, r);
```